fairmq/sdk/Topology.h

```diff
@@ -308,13 +321,7 @@ class BasicTopology : public AsioBase<Executor, Allocator>
         fDDSSession.SendCommand(cmd::Cmds(cmd::make<cmd::UnsubscribeFromStateChange>()).Serialize());
 
         // wait for all tasks to confirm unsubscription
-        std::unique_lock<std::mutex> lk(*fMtx);
-        fStateChangeUnsubscriptionCV->wait(lk, [&](){
-            unsigned int count = std::count_if(fStateIndex.cbegin(), fStateIndex.cend(), [=](const auto& s) {
-                return fStateData.at(s.second).subscribed_to_state_changes == false;
```

Comment:
I think with my changes here, we don't need the `subscribed_to_state_changes` member at all. But since it is part of the public API, I leave it in. Perhaps someone will even find it useful to have this info. Also, this member feels like an implementation detail to me.

Reply:
Why did you add the counter?

Reply:
It's faster and easier to check the counter when waiting for all to subscribe or unsubscribe, than to do a count_if each time the count changes. It's used both in ctor and dtor.

Reply:
Your question makes me think, what if a task erroneously sends a un/subscription message more than once. Then counter will be wrong as it is implemented now. I will add a check and warning in the SDK if that happens.

Reply:
I recommend to factor the count_if stuff into its own function and then use a counter as a "cache" type of thing. would also need a counterpart function to update the subscription status.

Reply:
Why? with the counter there is no need for count_if at all.

Reply:
Well, on the getter side, no, but you would still need `fStateData` to get your counter to count the right thing. So, the logic (that accesses the `fStateData`) would move to the setter side, right?

Reply:
If I understand your counter correctly, your argument is: When we need to query the amount of subscribers, let's re-implement it by not counting on every query, but by pre-populating the query result in a variable which we update on every update on the subscribers, right?

Reply:
standard space-time trade-off thingy

Reply:
yes